pkg/controller/machine-set-boot-image/machine_set_boot_image_controller.go

```diff
@@ -256,21 +263,19 @@ func (ctrl *Controller) updateMachineSet(oldMS, newMS interface{}) {
 	oldMachineSet := oldMS.(*machinev1beta1.MachineSet)
 	newMachineSet := newMS.(*machinev1beta1.MachineSet)
 
-	// Only take action if the machineset.Generation has bumped indicating a change in the resource has taken place.
-	// This callback seems to take place fairly frequently when there has been no changes in the resource itself
-	// and this check helps avoid unnecessary full loop syncs.
-	if oldMachineSet.Generation == newMachineSet.Generation {
+	// Only take action if the there is an actual change in the MachineSet's Spec, or a change in the labels
+	if oldMachineSet.Generation == newMachineSet.Generation && reflect.DeepEqual(oldMachineSet.Labels, newMachineSet.Labels) {
 		return
 	}
 
-	klog.Infof("MachineSet %s updated, reconciling all machinesets", oldMachineSet.Name)
+	klog.Infof("MachineSet %s updated, reconciling enrolled machineset resources", oldMachineSet.Name)
 
-	// Update all machinesets instead of just this once. This prevents needing to maintain a local
+	// Update all machinesets instead of just this one. This prevents needing to maintain a local
 	// store of machine set conditions. As this is using a lister, it is relatively inexpensive to do
 	// this.
-	err := ctrl.enqueueAllMachineSets()
+	err := ctrl.enqueueMAPIMachineSets()
 	if err != nil {
-		klog.Errorf("Error enqueuing all machine sets: %v", err)
+		klog.Errorf("Error enqueuing MAPI machine sets: %v", err)
```

Comment:
should this be `machinesets` ?

Reply:
Also, we may want to log machineset name that failed to enque

Reply:
Will fix those, I see a bunch of "machine set" lurking about! 

And sorry about the somewhat confusing names, to clarify:
- `enqueueMachineSets()` -> this checks the `MachineConfiguration.spec.managedBootImages` knob and calls `enqueueMachineSet()` on every enrolled machineset
-  `enqueueMachineSet()` -> adds a target machineset to the workqueue. The error that can happen here is a bad key, but seems unlikely, which is logged
- Anything in the queue will then go through syncMachineResource(), which should report any misbehaving machineset, and possibly cause a prometheus alert. We will update this to a condition on the MachineConfiguration object before we GA

Reply:
Thanks David for providing additional context! makes sense.